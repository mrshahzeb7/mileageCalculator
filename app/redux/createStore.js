import {applyMiddleware, compose, createStore} from 'redux';

export default (rootReducer, rootSaga) => {
  const middleware = [];
  const enhancers = [];

  enhancers.push(applyMiddleware(...middleware));

  const createAppropriateStore = createStore;

  const store = createAppropriateStore(rootReducer, compose(...enhancers));

  return {
    store,
  };
};
