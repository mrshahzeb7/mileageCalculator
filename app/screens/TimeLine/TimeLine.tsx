import React, {useEffect} from 'react';
import {StyleSheet, View} from 'react-native';
import {Loader, TimeLineList} from '../../components';
import {MainActions, useAppSelector} from '../../redux';
import {useDispatch} from 'react-redux';

interface Props {}

export const TimeLine: React.FC<Props> = () => {
  const {data, isLoading} = useAppSelector(state => state?.main?.timeLineList);
  const dispatch = useDispatch();

  const getData = () => {
    if (data?.length) {
      return;
    }
    setTimeout(() => {
      dispatch(MainActions.timeLineContentRequest());
    }, 1000);
  };

  useEffect(() => {
    getData();

    const timer = setTimeout(() => {
      // @ts-ignore

      data?.length && alert(`There are ${data?.length} items in this list`);
    }, 2000);
    return () => {
      clearTimeout(timer);
    };
  }, []);

  return (
    <View style={styles.container}>
      {isLoading ? <Loader /> : <TimeLineList data={data} />}
    </View>
  );
};

// styles
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
