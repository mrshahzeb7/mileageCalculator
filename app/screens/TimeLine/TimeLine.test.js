import React from 'react';
import 'react-native';

// Note: import explicitly to use the types shiped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import createStore from '../../redux';
import {TimeLine} from './TimeLine';
const store = createStore();

describe('TimeLine', () => {
  it('TimeLine renders correctly', () => {
    const tree = renderer.create(<Parent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
const Parent = () => {
  return (
    <Provider store={store}>
      <TimeLine />
    </Provider>
  );
};
