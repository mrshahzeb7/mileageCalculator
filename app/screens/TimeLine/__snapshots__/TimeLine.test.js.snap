// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLine TimeLine renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      color="#93b2fd"
      size="large"
    />
  </View>
</View>
`;
